packages/compass-components/src/components/editor.tsx

```diff
@@ -72,32 +75,54 @@ function Editor({
     ...(!!completer && { enableLiveAutocompletion: true }),
   };
 
+  const editorId = useId('ace-editor-');
+
   return (
-    <AceEditor
-      mode={
-        variant === 'Generic'
-          ? undefined
-          : variant === 'EJSON'
-          ? 'json'
-          : 'javascript' // set to 'mongodb' as part of setOptions
-      }
-      theme="mongodb"
-      width="100%"
-      value={text}
-      onChange={onChangeText}
-      editorProps={{ $blockScrolling: Infinity }}
-      setOptions={setOptions}
-      readOnly={readOnly}
-      {...aceProps}
-      onFocus={(ev: any) => {
-        if (completer) {
-          tools.setCompleters([completer]);
+    <div data-testid={dataTestId}>
+      <AceEditor
+        mode={
+          variant === 'Generic'
+            ? undefined
+            : variant === 'EJSON'
+            ? 'json'
+            : 'javascript' // set to 'mongodb' as part of setOptions
         }
-        onFocus?.(ev);
-      }}
-    />
+        theme="mongodb"
+        width="100%"
+        value={text}
+        onChange={onChangeText}
+        editorProps={{ $blockScrolling: Infinity }}
+        setOptions={setOptions}
+        readOnly={readOnly}
+        // name should be unique since it gets translated to an id
+        name={aceProps.name ?? editorId}
+        {...aceProps}
+        onFocus={(ev: any) => {
+          if (completer) {
+            tools.setCompleters([completer]);
+          }
+          onFocus?.(ev);
+        }}
+      />
+    </div>
   );
 }
 
+/**
+ * Sets the editor value, use this with RTL like this:
+ *
+ * ```
+ * render(<Editor data-testid='my-editor' />);
+ * setEditorValue(screen.getByTestId('editor-test-id'), 'my text');
+ * ```
+ */
+function setEditorValue(element: HTMLElement, value: string): void {
+  const container = element.querySelector('.ace_editor');
+  if (!container) {
+    throw new Error('Cannot find editor container');
+  }
+  (window as any).ace.edit(container.id).setValue(value);
+}
```

Comment:
Neat! I wonder if we can use this in e2e tests for mac as a workaround for broken clipboard in evergreen 🤔

Reply:
Ooh yes this should be helpful.